agent/src/workflows/post_twitter/workflow.py

```diff
@@ -0,0 +1,121 @@
+import os
+import random
+from logging import Logger
+import tweepy
+from pydantic import BaseModel
+from vellum import ChatMessagePromptBlock, JinjaPromptBlock, PromptParameters
+from vellum.workflows import BaseWorkflow
+from vellum.workflows.nodes import BaseNode, InlinePromptNode
+
+
+class TweetContent(BaseModel):
+    text: str
+    hashtags: list[str]
+
+
+def generate_tweets(tweets: list[TweetContent]) -> None:
+    pass
+
+
+class GenerateTweetContent(InlinePromptNode):
+    ml_model = "gpt-4o"
+    blocks = [
+        ChatMessagePromptBlock(
+            chat_role="SYSTEM",
+            blocks=[
+                JinjaPromptBlock(
+                    template="""\
+Generate engaging Twitter content for a personal/professional account. The content should be:
+- Authentic and personal
+- Valuable to followers (insights, tips, thoughts)
+- Appropriate for a software engineer/entrepreneur
+- Under 280 characters
+- Engaging and likely to drive interaction
+
+Create 3 different tweet options with different styles:
+1. A professional insight or tip
+2. A personal reflection or observation
+3. An industry trend or commentary
+
+For each tweet, provide:
+- Text (under 280 characters)
+- Hashtags (2-4 relevant hashtags)
+"""
+                )
+            ],
+        )
+    ]
+    functions = [
+        generate_tweets,
+    ]
+    parameters = PromptParameters(
+        temperature=0.8,
+        max_tokens=2000,
+    )
+
+
+class SelectTweet(BaseNode):
+    raw_tweets = GenerateTweetContent.Outputs.results[0]["value"]["arguments"]["tweets"]
+
+    class Outputs(BaseNode.Outputs):
+        selected_tweet = TweetContent
+
+    def run(self) -> Outputs:
+        selected_tweet = TweetContent.model_validate(random.choice(self.raw_tweets))
+        logger: Logger = getattr(self._context, "logger")
+        logger.info(f"Selected tweet: {selected_tweet.text}")
+        return self.Outputs(selected_tweet=selected_tweet)
+
+
+class PostToTwitter(BaseNode):
+    selected_tweet = SelectTweet.Outputs.selected_tweet
+
+    class Outputs(BaseNode.Outputs):
+        tweet_id: str
+        summary: str
+
+    def run(self) -> Outputs:
+        logger: Logger = getattr(self._context, "logger")
+        
+        api_key = os.getenv("TWITTER_API_KEY")
+        api_secret = os.getenv("TWITTER_API_SECRET")
+        access_token = os.getenv("TWITTER_ACCESS_TOKEN")
+        access_token_secret = os.getenv("TWITTER_ACCESS_TOKEN_SECRET")
```

Comment:
Instead of getting these env's from environment, fetch them from the Application's table